src/main/java/com/minecolonies/coremod/client/gui/WindowHutTavern.java

```diff
@@ -8,19 +8,21 @@
 import com.minecolonies.api.colony.ICitizenDataView;
 import com.minecolonies.api.util.constant.Constants;
 import com.minecolonies.coremod.Network;
-import com.minecolonies.coremod.colony.buildings.workerbuildings.BuildingHome;
-import com.minecolonies.coremod.colony.buildings.workerbuildings.BuildingTavern;
+import com.minecolonies.coremod.colony.buildings.modules.TavernBuildingModule;
+import com.minecolonies.coremod.colony.buildings.views.LivingBuildingView;
+import com.minecolonies.coremod.network.messages.server.colony.building.RecallCitizenHutMessage;
 import com.minecolonies.coremod.network.messages.server.colony.building.home.AssignUnassignMessage;
 import net.minecraft.client.Minecraft;
 import org.jetbrains.annotations.NotNull;
 
 import static com.minecolonies.api.util.constant.TranslationConstants.COM_MINECOLONIES_COREMOD_GUI_HOME_ASSIGN;
 import static com.minecolonies.api.util.constant.TranslationConstants.COM_MINECOLONIES_COREMOD_GUI_WORKERHUTS_LEVEL_0;
+import static com.minecolonies.api.util.constant.WindowConstants.BUTTON_RECALL;
 
 /**
  * Window for the tavern
  */
-public class WindowHutTavern extends AbstractWindowBuilding<BuildingTavern.View>
+public class WindowHutTavern extends AbstractWindowBuilding<TavernBuildingModule.View>
```

Comment:
How is this going to work when we have multiple modules that need to participate in the view? For instance, Crafter (job queue) and Smelter (Fuel)

Reply:
Yeah, I commented that in the description. That is the next PR I want to make. This one was getting too big already.

Reply:
Fair enough.